docs/development/workflow/development_workflow.rst

```diff
@@ -433,24 +433,42 @@ to GitHub. GitHub will automatically update your pull request.
 Rebase, but only if asked
 =========================
 
-Sometimes the maintainers of Astropy will ask you to *rebase* your changes
-before they are merged into the main Astropy repository.
+Sometimes the maintainers of Astropy may ask a pull request to be *rebased*
+(`git rebase`) or *squashed*/*squished* (`git merge --squash`) before
+changes in a Pull Request are merged into the main Astropy
```

Comment:
maybe rephrase along the lines of "when there is a conflict with the current master", but you already say that in the next paragraph.